services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/entity/VpcResourceMeta.java

```diff
@@ -0,0 +1,25 @@
+package com.futurewei.alcor.netwconfigmanager.entity;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class VpcResourceMeta {
+
+    private String vni;
+
+    // Private IP => List<ResourceMetadata>
+    private HashMap<String, List<ResourceMeta>> resourceMetas;
+
+    public VpcResourceMeta(String vni, HashMap<String, List<ResourceMeta>> resourceMetas) {
+        this.vni = vni;
+        this.resourceMetas = new HashMap<String, List<ResourceMeta>>(resourceMetas);
+    }
+
+    public String getVni() {
+        return this.vni;
+    }
+
+    public HashMap<String, List<ResourceMeta>> getResourceMetas() {
+        return this.resourceMetas;
+    }
+}
```

Comment:
It looks like there is no way to add more entries to the list of ResourceMeta for a given vni. So the entires list must be available when a VpcResourceMeta is constructed. Is this the use case? ResourceMeta is also of simliar design.

I think putting ResourceMeta followed by VpcResourceMeta at the beginning would be easier to follow the code.

I think these classes form a rather complex nested hashes that a simple picturization will be a great help to understand it.

Reply:
> It looks like there is no way to add more entries to the list of ResourceMeta for a given vni. So the entires list must be available when a VpcResourceMeta is constructed. Is this the use case? ResourceMeta is also of simliar design.

We just update ResourceMeta with a more flexible data structure and allow field editing.. more to come.